### PR TITLE
Chore/validation

### PR DESCRIPTION
**Pull request by pwright08**
- added express-validator to package.json in internal-web
- added middleware to server.js
- added sample validation to about-you page on title, name and DOB fields

Associated ODP documentation can be found [here](https://github.com/KainosSoftwareLtd/gov-odp/wiki/moj-noms-apvs-chore-validation).

### CODE REVIEW

File: alpha/external-web/app/routes/about-you.js [hunk 1]
```diff
@@ -87,3 +99,20 @@ function updateEligibilityFlag (id, newID) {
       })
   }
 }
+
+function validationErrors (request, response, next) {
```
Comment (stevenalexander) [inline alpha/external-web/app/routes/about-you.js]: split out into validator-about-you.js which is included as a require and called in post

File: alpha/external-web/app/routes/about-you.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ var router = require('../routes')
 var client = require('../services/eligibility-client')
 var eligibilityFlag = require('../services/eligibility-flag')
 var logger = require('../services/bunyan-logger')
+var util = require('util')
```
Comment (stevenalexander) [inline alpha/external-web/app/routes/about-you.js]: why is this needed?

File: alpha/external-web/server.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ var bodyParser = require('body-parser')
 var browserSync = require('browser-sync')
 var config = require('./app/config.js')
 var utils = require('./lib/utils.js')
+var expressValidator = require('express-validator')
```
Comment (stevenalexander) [inline alpha/external-web/server.js:12]: as discussed, might be better to use base validator library rather than include as middleware

File: alpha/external-web/app/routes/about-you.js (hunk 2 above)
Comment (pwright08) [reply alpha/external-web/app/routes/about-you.js]: I don't think it's needed, it was used in the example on [express-validator github](https://github.com/ctavan/express-validator) but I think we can use JSON.stringify instead

File: alpha/external-web/app/routes/about-you.js [hunk 4]
```diff
@@ -23,11 +24,19 @@ router.get('/about-you/:claimant_id', function (request, response, next) {
...
+  if (validator.validationErrors(request, response, next)) {
+    return
+  }
+
   save(null, request, response)
   next()
 })
 
 router.post('/about-you/:claimant_id', function (request, response, next) {
+  if (validator.validationErrors(request, response, next)) {
```
Comment (stevenalexander) [inline alpha/external-web/app/routes/about-you.js:36]: there is no reason to pass in response/next, this is a synchronous function

File: alpha/external-web/app/services/validators/about-you-validator.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    .notEmpty().isNumeric()
+  request.checkBody('dob-month', 'dob-month ' + NOT_EMPTY_MESSAGE + AND + IS_NUMERIC_MESSAGE)
+    .notEmpty().isNumeric()
+  request.checkBody('dob-year', 'dob-year ' + NOT_EMPTY_MESSAGE + AND + IS_NUMERIC_MESSAGE)
+    .notEmpty().isNumeric()
+
+  var errors = request.validationErrors(true)
+
+  if (errors) {
+    response.status(400).render('error', { error: VALIDATION_ERROR_MESSAGE + JSON.stringify(errors) })
```
Comment (stevenalexander) [inline alpha/external-web/app/services/validators/about-you-validator.js:24]: validator should just return the errors, not control the response

File: alpha/external-web/server.js [hunk 6]
```diff
@@ -65,6 +66,8 @@ app.use(bodyParser.json())
 app.use(bodyParser.urlencoded({
   extended: true
 }))
+// Support for express-validator
+app.use(expressValidator())
```
Comment (stevenalexander) [inline alpha/external-web/server.js:70]: want to check again if we need to use express-validator over validator, as we do not need to use it as middleware
